### PR TITLE
normalizing lock addresses in store

### PR DESCRIPTION
**Pull request by julien51**
# Description

This will avoid adding duplicate locks to the store when the addresses are dlightly different (case).

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/__tests__/reducers/locksReducer.test.js [hunk 1]
```diff
@@ -49,12 +49,15 @@ describe('locks reducer', () => {
...
-  it('should add the lock when receiving CREATE_LOCK and if it was not there yet', () => {
+  it('should add the lock with a nornalized address when receiving CREATE_LOCK and if it was not there yet', () => {
     expect.assertions(1)
     expect(
       reducer(undefined, {
         type: CREATE_LOCK,
-        lock,
+        lock: {
+          ...lock,
+          address: lock.address.toUpperCase(),
```
Comment (cnasc) [inline unlock-app/src/__tests__/reducers/locksReducer.test.js]: It looks like in the code we normalize by using `.toLowerCase()`, is this difference expected?

Comment (julien51) [reply unlock-app/src/__tests__/reducers/locksReducer.test.js]: Yes that's the test ;) I am testing that sending an uppercase address will get normalized into a lowercase...

**Comment by julien51** [comment]
interesting failure on the integration test. Let me dig deeper.

**Comment by julien51** [comment]
Ok, I found the issue but this is kind of a mess. I'll refactor the lock actions first.

**Comment by julien51** [comment]
Done, please, take a look when you have a moment!

**Comment by julien51** [comment]
Grrrr. This worked yesterday at some point. This is a timing issue. I'll keep digging

**Pull request closed without merging**